examples/deepspeed/cifar10_cpu_offloading/README.md

```diff
@@ -0,0 +1,105 @@
+# DeepSpeed CPU Offloading example
+This example is adapted from the 
+[CIFAR example in the DeepSpeedExamples](https://github.com/microsoft/DeepSpeedExamples/tree/master/cifar) 
+repository. It is intended to show how to configure 
+[ZeRO Stage 3 with CPU offloading](https://www.deepspeed.ai/tutorials/zero/) for a simple CNN network.
+
+Compared to the original example, MoE is removed due to `AssertionError: MoE not supported with Stage 3`.
+
+**IMPORTANT!** 
+ZeRO Stage 3 enables training models that do not fit into GPU memory by offloading optimizer, parameters and gradients to CPU
+(RAM memory). The assumption is that RAM memory >> GPU memory, however when running on an instance with limited RAM 
+memory, it is possible to experience memory-related errors as there is not enough memory for offloading.
+Read more about memory requirements at https://deepspeed.readthedocs.io/en/latest/memory.html and take a look ath the 
+`Final notes` section at the end.
+
+
+## Files
+* **model_def_OOM_no_offload.py**: The core code for the model implemented as DeepSpeedTrial. Running this model without 
+ZeRO Infinity CPU offload results in CUDA Out of Memory (OOM) error.
+* **model_def_OOM_offload.py**: The same model architecture as in `model_def_OOM_no_offload.py`, however this time the 
+model is running with ZeRO Stage 3 CPU offloading. 
+The key difference is initializing DeepSpeed ZeRO in line 50:
+```
+with deepspeed.zero.Init():
+    model = Net(self.args)
+```
+`deepspeed.zero.Init()` creates a context that accelerates model initialization and allows to allocate models that
+exceed the size of local CPU/GPU memory/NVMe, but fit within the aggregated CPU/GPU memory/NVMe across all nodes. 
+
+Initializing a model in the `deepspeed.zero.Init()` context has the following effects:
+* allocates tensors to CPU or GPU memory or NVMe,
+* converts floating points tensors to half precision,
+* **immediately** partitions tensors among the group of data-parallel devices,
+* _(optional)_ uses more memory-efficient Linear Layer implementation.
+
+Note that even if you provide ZeRO Stage 3 configuration, but the model is not initialized in the context, the model 
+will not be partitioned.
```

Comment:
i believe the model weights, gradients, and optimizer state are all eventually partitioned fro ZeRO 3 just not during initialization if model is not created in the deepspeed.zero.Init context.

maybe change to something like "...but the model is not initialized in the context, the full model will have to fit into CPU memory in order to be initialized correctly."

Reply:
You are correct. It will be partitioned eventually, and my writing is misleading in that sense. Fixing now.